test/headers.js

```diff
@@ -145,9 +145,9 @@ test('remove null value headers', async t => {
 test('remove undefined value headers', async t => {
```

Comment:
Wait. What's with `user-agent`? You need to change the test name if you change the behaviour too.